### ISSUE TITLE
Training BUG

### ISSUE THREAD

**Issue by MeycL** [HKUDS/DCRec#2]
Thank you very much for your wonderful work. When I have Training loss is nan when I train, how should I solve this problem?

**Comment by yuh-yang** [comment]
Hi,

Can you try our updated commit? Sometimes we observe NaN loss because of the distribution restrain loss turning NaN, when in some cases the generated normal distribution contains negative values. We developed two versions of computing strategies to avoid this, which are updated in the newest commit:

```
def cal_kl(target, input):
    ### log with sigmoid
    target = torch.sigmoid(target)
    input = torch.sigmoid(input)
    target = torch.log(target + 1e-8)
    input = torch.log(input + 1e-8)
    return F.kl_div(input, target, reduction='batchmean', log_target=True)

def cal_kl_1(target, input):
    target[target<1e-8] = 1e-8
    target = torch.log(target + 1e-8)
    input = torch.log_softmax(input + 1e-8, dim=0)
    return F.kl_div(input, target, reduction='batchmean', log_target=True)
```
